misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java

```diff
@@ -920,6 +956,645 @@ public NilTypeDescriptorNode transform(NilTypeDescriptorNode nilTypeDescriptorNo
+        TypeDescriptorNode typeNode = formatNode(typeParameterNode.typeNode(), 0, 0);
+        Token gtToken = formatToken(typeParameterNode.gtToken(), this.trailingWS, this.trailingNL);
+
+        return typeParameterNode.modify()
+                .withTypeNode(typeNode)
+                .withLtToken(ltToken)
+                .withGtToken(gtToken)
+                .apply();
+    }
+
+    @Override
+    public FunctionTypeDescriptorNode transform(FunctionTypeDescriptorNode functionTypeDescriptorNode) {
+        NodeList<Token> qualifierList = formatNodeList(functionTypeDescriptorNode.qualifierList(), 1, 0, 0, 1);
+        Token functionKeyword = formatToken(functionTypeDescriptorNode.functionKeyword(), 1, 0);
+        FunctionSignatureNode functionSignature = formatNode(functionTypeDescriptorNode.functionSignature(),
+                this.trailingWS, this.trailingNL);
+
+        return functionTypeDescriptorNode.modify()
+                .withQualifierList(qualifierList)
+                .withFunctionKeyword(functionKeyword)
+                .withFunctionSignature(functionSignature)
+                .apply();
+    }
+
+    @Override
+    public ParenthesisedTypeDescriptorNode transform(ParenthesisedTypeDescriptorNode parenthesisedTypeDescriptorNode) {
+        Token openParenToken = formatToken(parenthesisedTypeDescriptorNode.openParenToken(), 0, 0);
+        TypeDescriptorNode typedesc = formatNode(parenthesisedTypeDescriptorNode.typedesc(), 0, 0);
+        Token closeParenToken = formatToken(parenthesisedTypeDescriptorNode.closeParenToken(),
+                this.trailingWS, this.trailingNL);
+
+        return parenthesisedTypeDescriptorNode.modify()
+                .withOpenParenToken(openParenToken)
+                .withTypedesc(typedesc)
+                .withCloseParenToken(closeParenToken)
+                .apply();
+    }
+
+    @Override
+    public ExternalFunctionBodyNode transform(ExternalFunctionBodyNode externalFunctionBodyNode) {
+        Token equalsToken = formatToken(externalFunctionBodyNode.equalsToken(), 1, 0);
+        NodeList<AnnotationNode> annotations = formatNodeList(externalFunctionBodyNode.annotations(), 0, 1, 1, 0);
+        Token externalKeyword = formatToken(externalFunctionBodyNode.externalKeyword(), 0, 0);
+        Token semicolonToken = formatToken(externalFunctionBodyNode.semicolonToken(), this.trailingWS, this.trailingNL);
+
+        return externalFunctionBodyNode.modify()
+                .withEqualsToken(equalsToken)
+                .withAnnotations(annotations)
+                .withExternalKeyword(externalKeyword)
+                .withSemicolonToken(semicolonToken)
+                .apply();
+    }
+
+    @Override
+    public AnnotationNode transform(AnnotationNode annotationNode) {
+        Token atToken = formatToken(annotationNode.atToken(), 0, 0);
+        Node annotReference = formatNode(annotationNode.annotReference(), 1, 0);
```

Comment:
if annotationNode.annotValue() is not present, trailing WS and NL should get from the top